__tests__/main.test.ts

```diff
@@ -121,3 +129,125 @@ test('does not display module information when it does not exist', async () => {
 
   expect(printChunkModulesTable(statsDiff)).toMatchSnapshot()
 })
+
+describe('printAssetTablesByGroup describes asset sections as requested', () => {
+  // generate all combinations of sections
+  const cases: DescribeAssetsOptions[] = []
+  for (let i = 0; i < Math.pow(2, describeAssetsSections.length); i++) {
+    const options = {} as DescribeAssetsOptions
+    for (let n = 0; n < describeAssetsSections.length; n++) {
+      if ((i >> n) & 1) {
+        options[describeAssetsSections[n]] = true
+      } else {
+        options[describeAssetsSections[n]] = false
+      }
+    }
+    cases.push(options)
+  }
+
+  let statsDiff: WebpackStatsDiff
+  beforeAll(async () => {
+    statsDiff = getStatsDiff(
+      await readJsonFile('./__mocks__/old-stats-assets.json'),
+      await readJsonFile('./__mocks__/new-stats-assets.json')
+    )
+  })
+
+  test.each(cases)(
+    'printAssetTablesByGroup: %j',
+    (options: DescribeAssetsOptions) => {
+      const assetTables = printAssetTablesByGroup(statsDiff, options)
+      for (const [section, included] of Object.entries(options)) {
+        const sectionHeader = `**${section[0].toUpperCase()}${section.slice(
+          1
+        )}**`
+        if (included) {
+          expect(assetTables).toContain(sectionHeader)
+        } else {
+          expect(assetTables).not.toContain(sectionHeader)
+        }
+      }
+      if (Object.entries(options).every(([, included]) => included === false)) {
+        expect(assetTables).toBe('')
+      }
+    }
+  )
+})
+
+describe('getDescribeAssetsOptions', () => {
+  test(`getDescribeAssetsOptions: "all"`, () => {
+    const generatedOptions = getDescribeAssetsOptions('all')
+    for (const section of describeAssetsSections) {
+      expect(generatedOptions[section]).toBe(true)
+    }
+  })
+
+  test(`getDescribeAssetsOptions: "none"`, () => {
+    const generatedOptions = getDescribeAssetsOptions('none')
+    for (const section of describeAssetsSections) {
+      expect(generatedOptions[section]).toBe(false)
+    }
+  })
+
+  test(`getDescribeAssetsOptions: "changed-only"`, () => {
+    const generatedOptions = getDescribeAssetsOptions('changed-only')
+    for (const section of describeAssetsSections) {
+      if (section === 'unchanged') {
+        expect(generatedOptions[section]).toBe(false)
+      } else {
+        expect(generatedOptions[section]).toBe(true)
+      }
+    }
+  })
+
+  test('getDescribeAssetsOptions: handles keyword with spaces', () => {
+    const generatedOptions = getDescribeAssetsOptions('   all  ')
+    for (const section of describeAssetsSections) {
+      expect(generatedOptions[section]).toBe(true)
+    }
+  })
+
+  test('getDescribeAssetsOptions: unsupported option throws', () => {
+    try {
+      getDescribeAssetsOptions('unsupported options')
+      fail('should fail with unsupported options')
+    } catch (e) {
+      // pass
+    }
```

Comment:
🎨 would it make sense to use [`expect(...).toThrow()`](https://jestjs.io/docs/expect#tothrowerror) here?

```suggestion
    expect(() => getDescribeAssetsOptions('unsupported options')).toThrow()
```